File: typescript/packages/management/src/tools/list_agents.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const listAgents: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  return ctx.client.agents.list(ctx.sandboxId, {
    status:
      input.status === 'active' || input.status === 'archived'
        ? input.status
        : undefined,
  });
};

File: python/copass-management/src/copass_management/tools/list_agents.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def list_agents(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    status = input.get("status")
    return await ctx.client.agents.list(
        ctx.sandbox_id,
        status=status if status in ("active", "archived") else None,
    )
